### ISSUE TITLE
Debug node and LSL meta is too verbose

### ISSUE THREAD

**Issue by dojeda** [timeflux/timeflux#39]
Since https://github.com/timeflux/timeflux/commit/f2984fb5dfd37f3383f7343ced70aa7e9fe28073 the debug node displays meta information as well, which is very useful.
However, since https://github.com/timeflux/timeflux/commit/0393019a85f09966cb2ca1e59686014407f5511c the lsl sends the received stream information as meta and  somehow this is sent every time, making the logs difficult to read with unnecessary information.

I am not sure if sending the meta all the time is a bug. 

I have an alternative debug node that silences meta with a parameter, and defaults to not showing the meta (as it was done previously):

```python
from timeflux.core.node import Node


class Display(Node):
    """Display input."""

    def __init__(self, show_input=True, show_meta=False):
        super().__init__()
        self._show_input = show_input
        self._show_meta = show_meta

    def update(self):
        if self._show_input and self.i.ready():
            self.logger.debug('\n%s', self.i.data)
        if self._show_meta and self.i.meta:
            self.logger.debug('\n%s', self.i.meta)
```

By the way, the logging level could be dynamic:
```python
import logging

from timeflux.core.node import Node


class Display(Node):
    """Display input."""

    def __init__(self, level='DEBUG', show_input=True, show_meta=False):
        super().__init__()
        self._show_input = show_input
        self._show_meta = show_meta
        self._level = getattr(logging, level, logging.NOTSET)

    def update(self):
        if self._show_input and self.i.ready():
            self.logger.log(self._level, '\n%s', self.i.data)
        if self._show_meta and self.i.meta:
            self.logger.log(self._level, '\n%s', self.i.meta)
```

Would it be interesting to merge in this modification on the debug node, independently from the question regarding the constant meta propagation of the lsl node?

**Comment by mesca** [comment]
I fully agree with `show_input` and `show_meta`. Please feel free to open a PR.
I am unsure about the custom level param, since this is... well, a debug node ;) But if you have a specific use case in mind, let me know.
Another useful modification on this node would be to accept multiple inputs in addition to the default one (and obviously provide multiple outputs).

**Comment by dojeda** [comment]
> I am unsure about the custom level param, since this is... well, a debug node ;) But if you have a specific use case in mind, let me know.

Yes, here is one: we often acquire and save events from another program, like unity. It is useful to keep a debug node printing the events received, but keeping the level on DEBUG will show a lot of messages from other places, like congestions, etc. Ideally, I want to keep showing events on the console, but not the whole debug level messages.

> Another useful modification on this node would be to accept multiple inputs in addition to the default one (and obviously provide multiple outputs).

Agreed, this would reduce the number of debug nodes. But if the implementation is not as straightforward as a `for x in self.inputs: self.logger.log(...,x)` (because of the meta, for example), then can we do it in a separate issue?

**Comment by mesca** [comment]
> Yes, here is one: we often acquire and save events from another program, like unity. It is useful to keep a debug node printing the events received, but keeping the level on DEBUG will show a lot of messages from other places, like congestions, etc. Ideally, I want to keep showing events on the console, but not the whole debug level messages.

Sold.

> Agreed, this would reduce the number of debug nodes. But if the implementation is not as straightforward as a `for x in self.inputs: self.logger.log(...,x)` (because of the meta, for example), then can we do it in a separate issue?

Sure.

**Comment by mesca** [comment]
> > Another useful modification on this node would be to accept multiple inputs in addition to the default one (and obviously provide multiple outputs).
> 
> Agreed, this would reduce the number of debug nodes. But if the implementation is not as straightforward as a `for x in self.inputs: self.logger.log(...,x)` (because of the meta, for example), then can we do it in a separate issue?

Let's track this in #47.

**Comment by mesca** [comment]
The verbosity has been addressed in #44.

**Comment by mesca** [comment]
@dojeda Feel free to submit a PR for the custom logging level in the debug node.
